assertion/function/assertiontree/backprop_util.go

```diff
@@ -749,3 +750,22 @@ func CheckGuardOnFullTrigger(trigger annotation.FullTrigger) annotation.FullTrig
 	}
 	return trigger
 }
+
+// addAssignmentToConsumer updates the consumer with assignment entries for informative printing of errors
+func addAssignmentToConsumer(lhs, rhs ast.Expr, pass *analysis.Pass, consumer annotation.ConsumingAnnotationTrigger) (err error) {
+	var lhsExprStr, rhsExprStr string
+	if lhsExprStr, err = asthelper.PrintExpr(lhs, pass, true /* isShortenExpr */); err != nil {
+		return err
+	}
+	if rhsExprStr, err = asthelper.PrintExpr(rhs, pass, true /* isShortenExpr */); err != nil {
+		return err
+	}
+
+	consumer.AddAssignment(annotation.Assignment{
```

Comment:
Not something to be handled in this PR, but do you think we can actually move the logic (the `asthelper.PrintExpr`) inside the `AddAssignment` method? meaning make it take the AST nodes and do the expr shortening inside. I feel that would make the code simpler, but I may have missed some context.

Reply:
We could. However, that would mean passing the `*analysis.Pass` to `ConsumeTrigger`, which is not required by ConsumeTrigger otherwise. Maybe we can do this refactor later if that seems to be the better approach.